File: frontend/src/app/core/services/api-endpoints.ts
import { API_BASE_URL } from './api-base';

export const API_ENDPOINTS = {
  auth: {
    login: `${API_BASE_URL}/auth/login`,
    google: `${API_BASE_URL}/auth/google`,
    register: `${API_BASE_URL}/auth/register`,
    me: `${API_BASE_URL}/auth/me`,
    changePassword: `${API_BASE_URL}/auth/change-password`,
    avatar: `${API_BASE_URL}/auth/avatar`,
    refresh: `${API_BASE_URL}/auth/refresh`
  },
  plans: {
    collection: `${API_BASE_URL}/plans`,
    import: `${API_BASE_URL}/plans/import`,
    detail: (planId: number) => `${API_BASE_URL}/plans/${planId}`
  },
  dashboard: {
    summary: `${API_BASE_URL}/dashboard/summary`
  },
  strava: {
    status: `${API_BASE_URL}/integrations/strava/status`,
    activities: `${API_BASE_URL}/integrations/strava/activities`,
    exchangeToken: `${API_BASE_URL}/integrations/strava/exchange-token`,
    sync: `${API_BASE_URL}/integrations/strava/sync`,
    activityDetail: (activityId: number) =>
      `${API_BASE_URL}/integrations/strava/activities/${activityId}`
  }
} as const;

File: travans_mobile/lib/core/networking/api_endpoints.dart
class ApiEndpoints {
  const ApiEndpoints._();

  static const auth = _AuthEndpoints._();
  static const plans = _PlansEndpoints._();
  static const dashboard = _DashboardEndpoints._();
  static const strava = _StravaEndpoints._();
}

class _AuthEndpoints {
  const _AuthEndpoints._();

  final String login = '/auth/login';
  final String google = '/auth/google';
  final String register = '/auth/register';
  final String me = '/auth/me';
  final String refresh = '/auth/refresh';
  final String changePassword = '/auth/change-password';
  final String avatar = '/auth/avatar';
}

class _PlansEndpoints {
  const _PlansEndpoints._();

  final String collection = '/plans';
  final String import = '/plans/import';

  String detail(int planId) => '/plans/$planId';
}

class _DashboardEndpoints {
  const _DashboardEndpoints._();

  final String summary = '/dashboard/summary';
}

class _StravaEndpoints {
  const _StravaEndpoints._();

  final String status = '/integrations/strava/status';
  final String activities = '/integrations/strava/activities';
  final String exchangeToken = '/integrations/strava/exchange-token';
  final String sync = '/integrations/strava/sync';

  String activityDetail(int activityId) =>
      '/integrations/strava/activities/$activityId';
}
